synapse/storage/databases/main/events.py

```diff
@@ -1033,7 +1031,7 @@ def _persist_transaction_ids_txn(
         # With MSC3970, we rely on the device_id instead to scope the txn_id for events.
         # We're only inserting if MSC3970 is *enabled*, because else the pre-MSC3970
         # behaviour would allow for a UNIQUE constraint violation on this table
-        if to_insert_device_id and self._msc3970_enabled:
```

Comment:
Should we rip out storing the token ID now?

Reply:
I'm pretty sure we can. I guess it's only interesting to keep it in case of a rollback

Reply:
I put a TODO for now, I think we'll want to allow rollback (unless this isn't a big deal if it doesn't work...but we probably want to clean-up database table columns and such?)

Reply:
It looks like [`_cleanup_old_transaction_ids`](https://github.com/matrix-org/synapse/blob/2ad91ec628126753590c1a90c432270d6c8fa8fd/synapse/storage/databases/main/events_worker.py#L2093) will delete any transaction IDs after 24 hours. I think we probably want to keep it for now and after a version or two remove it.

Does that sound reasonable?

Reply:
If we choose to not rely on it anymore, sure